pipit/trace.py

```diff
@@ -29,3 +32,83 @@ def from_hpctoolkit(dirname):
         from .readers.hpctoolkit_reader import HPCToolkitReader
 
         return HPCToolkitReader(dirname).read()
+
+    @staticmethod
+    def from_nsight(dirname):
+        """Read an NSight trace into a new Trace object."""
+        # import this lazily to avoid circular dependencies
+        from .readers.nsight_reader import NSightReader
+
+        return NSightReader(dirname).read()
+
+    def binary(self, arr, low, high, x, start):
```

Comment:
Given a value and some intervals, is this trying to find which interval the value belongs within? I think a binary search might be unnecessary here.